packages/app-project/src/components/AuthModal/shared/components/withCustomFormik/withCustomFormik.spec.js

```diff
@@ -60,32 +60,33 @@ describe('Higher Order Component > withCustomFormik', function () {
     expect(formikHOC.props().validate).to.equal(validate)
   })
 
-  it('should call onChange prop', function () {
-    const onChangeStub = sinon.stub()
+  it('should call the custom change handler on change', function () {
 
+    const onChangeStub = sinon.stub()
     const wrapper = mount(
       <WrappedMockForm
         initialValues={initialValues}
         onChange={onChangeStub}
         onSubmit={onSubmit}
       />
     )
-    const formikProps = wrapper.find(MockForm).props()
-    let testInput = wrapper.find('input[name="testInput"]')
     const eventMock = {
       target: {
-        name: testInput.props().name,
+        name: 'testInput',
         value: 'foo'
       }
     }
+    const renderForm = wrapper.find('Formik').props().children
+    const mockInnerProps = {
+      values: {
+        testInput: 'foo'
+      },
+      handleChange: sinon.stub()
+    }
 
-    expect(testInput.props().value).to.equal('')
-    testInput.simulate('change', eventMock)
-    // diry chai does not work with sinon chai
-    expect(onChangeStub.withArgs(eventMock, formikProps)).to.have.been.calledOnce
-    testInput = wrapper.find('input[name="testInput"]')
-    // Formik's handleChange updates the input value for us based on the name attribute
-    // Test that the internal Formik handleChange has been called and worked
-    expect(testInput.props().value).to.equal('foo')
+    const mockForm = renderForm(mockInnerProps)
```

Comment:
Should I write this in JSX to make it clear what’s happening here?

Reply:
Yeah I think that'd help. I'm having a hard time understanding how we're testing this all.. maybe I shouldn't bother with the customization and just use the vanilla `withFormik` they provide?

Reply:
I’ll rewrite this in JSX tomorrow. It’s a quick change but I had a hard time figuring this out until I realised that Formik expects a component declaration, not a component instance, as a child. 

I’ve been expecting it to wrap a component instance, because that’s more intuitive to me in JSX. Passing a function as `props.chlidren` made it harder to reason about.

If `renderForm` is renamed to `InnerForm` then this line becomes `mockForm = <InnerForm {...mockInnerProps} />`.